auth_mellon_config.c

```diff
@@ -1139,6 +1143,52 @@ static const char *am_set_redirect_domains(cmd_parms *cmd,
     return NULL;
 }
 
+/* This function handles the MellonSignatureMethod configuration directive.
+ * This directive can be set to one of:
+ *
+ * Parameters:
+ *  cmd_parms *cmd       The command structure for this configuration
+ *                       directive.
+ *  void *struct_ptr     Pointer to the current directory configuration.
+ *  const char *arg      The string argument following this configuration
+ *                       directive in the configuraion file.
+ *
+ * Returns:
+ *  NULL on success or an error string if the argument is wrong.
+ */
+static const char *am_set_signature_method_slot(cmd_parms *cmd,
+                                                void *struct_ptr,
+                                                const char *arg)
+{
+    am_dir_cfg_rec *d = (am_dir_cfg_rec *)struct_ptr;
+
+    if (!strcasecmp(arg, "rsa-sha1")) {
+        d->signature_method = LASSO_SIGNATURE_METHOD_RSA_SHA1;
+    } else if (!strcasecmp(arg, "dsa-sha1")) {
+        d->signature_method = LASSO_SIGNATURE_METHOD_DSA_SHA1;
+    } else if (!strcasecmp(arg, "hmac-sha1")) {
+        d->signature_method = LASSO_SIGNATURE_METHOD_HMAC_SHA1;
+    } else if (!strcasecmp(arg, "rsa-sha256")) {
+        d->signature_method = LASSO_SIGNATURE_METHOD_RSA_SHA256;
+    } else if (!strcasecmp(arg, "hmac-sha256")) {
+        d->signature_method = LASSO_SIGNATURE_METHOD_HMAC_SHA256;
+    } else if (!strcasecmp(arg, "rsa-sha384")) {
+        d->signature_method = LASSO_SIGNATURE_METHOD_RSA_SHA384;
+    } else if (!strcasecmp(arg, "hmac-sha384")) {
+        d->signature_method = LASSO_SIGNATURE_METHOD_HMAC_SHA384;
+    } else if (!strcasecmp(arg, "rsa-sha512")) {
+        d->signature_method = LASSO_SIGNATURE_METHOD_RSA_SHA512;
+    } else if (!strcasecmp(arg, "hmac-sha512")) {
+        d->signature_method = LASSO_SIGNATURE_METHOD_HMAC_SHA512;
```

Comment:
I see that Travis (which is based on Ubuntu 14.04) fails due to missing the last 5 variants here (from HMAC_SHA256 and onwards). I therefore think they need to be behind a config test and `#ifdef`-guard.

Reply:
Interesting. All these signature methods were added in Lasso 2.5.0 (committed on 2/9/2015) including RSA_SHA256, I don't know how RSA_SHA256 was defined but not the others unless the Ubuntu version was patched. Anyway, that's not an issue. I've added tests inside configure.ac to test for the RSA_* variants (see below) and appropriate #if tests based on the result of the configure tests in the code. FWIW these constants are enums and cannot be tested with #ifdef, hence the need for the checks done in configure.